client_czmq.go

```diff
@@ -94,13 +96,22 @@ func (c *czmqSocketClient) sendChannel() chan *message {
 	return c.toMaster
 }
 
+func (c *gomqSocketClient) sendStatsChannel() chan *message {
+	return c.statsToMaster
+}
+
 func (c *czmqSocketClient) send() {
 	for {
 		select {
 		case <-c.shutdownChan:
 			return
 		case msg := <-c.toMaster:
-			c.sendMessage(msg)
+			go c.sendMessage(msg)
```

Comment:
It might just be good enough to use a goroutine to send the message -- it's likely the `serialize` call that's taking a long time

Reply:
We need to make sure the zeromq socket is thread safe. Have you tested the serialization of big message?